galaxy_ng/app/tasks/promotion.py

```diff
@@ -1,92 +1,161 @@
-        pk__in=repo.content.values_list("pk", flat=True)
-    ).values_list("pk", flat=True)
 
+def call_auto_approve_task(collection_version, repo):
+    """
+    Dispatches the auto approve task
+    """
     task_group = TaskGroup.current()
 
-    add_content_task = dispatch(
-        add_content,
+    auto_approve_task = dispatch(
+        auto_approve,
         exclusive_resources=[repo],
         task_group=task_group,
         kwargs=dict(
-            collection_version_pk=collection_version.pk,
-            repo_pk=repo.pk,
-            signatures_pks=list(signatures_pks),
+            cv_pk=collection_version.pk,
+            src_repo_pk=repo.pk,
         ),
     )
 
     task_group.finish()
 
-    return add_content_task
+    return auto_approve_task
 
 
 def call_move_content_task(collection_version, source_repo, dest_repo):
-    """Dispatches the move content task
-    This is a wrapper to group copy_content and remove_content tasks
-    because those 2 must run in sequence ensuring the same locks.
     """
-
-    signatures_pks = CollectionVersionSignature.objects.filter(
-        signed_collection=collection_version.pk,
-        pk__in=source_repo.content.values_list("pk", flat=True)
-    ).values_list("pk", flat=True)
+    Dispatches the move collection task
+    """
 
     return dispatch(
-        move_content,
+        move_collection,
         exclusive_resources=[source_repo, dest_repo],
         kwargs=dict(
-            collection_version_pk=collection_version.pk,
-            source_repo_pk=source_repo.pk,
-            dest_repo_pk=dest_repo.pk,
-            signatures_pks=list(signatures_pks),
+            cv_pk=collection_version.pk,
+            src_repo_pk=source_repo.pk,
+            dest_repo_list=[dest_repo.pk],
         ),
     )
 
 
-def add_content(collection_version_pk, repo_pk, signatures_pks=None):
-    """Import collection version + signatures to repository"""
-
-    content = [collection_version_pk]
-    if signatures_pks:
-        content += signatures_pks
-
-    # add content to the repo
-    add_and_remove(
-        repo_pk,
-        add_content_units=content,
-        remove_content_units=[],
-    )
-
+def copy_collection(cv_pk, src_repo_pk, dest_repo_list):
+    """
+    Copy a collection and all of it's contents into a list of destination repositories
+    """
+    src_repo = AnsibleRepository.objects.get(pk=src_repo_pk)
+    collection_version = CollectionVersion.objects.get(pk=cv_pk)
+
+    content_types = src_repo.content.values_list('pulp_type', flat=True).distinct()
+    source_pks = src_repo.content.values_list("pk", flat=True)
+
+    content = [cv_pk]
+
+    # collection signatures
+    if 'ansible.collection_signature' in content_types:
+        signatures_pks = CollectionVersionSignature.objects.filter(
+            signed_collection=cv_pk,
+            pk__in=source_pks
+        ).values_list("pk", flat=True)
+        if signatures_pks:
+            content.append(*signatures_pks)
```

Comment:
This will not work I guess

```python
In [4]: content = []

In [5]: t = [1,2,3]

In [6]: content.append(*t)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [6], in <cell line: 1>()
----> 1 content.append(*t)

TypeError: append() takes exactly one argument (3 given)

```

Must use extend method

```suggestion
            content.extend(signatures_pks)
```